src/ui/components/components.rs

```diff
@@ -9,10 +10,10 @@ use crate::runtime;
 
 
 pub trait Component<T> {
-    fn test(&mut self, info: Vec<T>);
+    fn load(&mut self, info: Vec<T>);
```

Comment:
The name is too generic `load_data` seems a better fit. However that will imply that the previous data is being erased which isn't what is happening in the implantation of this function so this trait should probably be expanded with `remove_old_data` function, and this function should be renamed to `load_new_data` if you will decided to go this route.